### PR TITLE
[WIP] Emoji rendering and addition

### PR DESCRIPTION
**Pull request by knrt10**
This PR will try to add emoji feature to live chat

- [x] Render emoji 
- [x] Add custom emojis
- [ ] Add emoji picker frame

### CODE REVIEW

**Comment by renatobecker** [comment]
@knrt10, there are some emojis not supported, like `:slight_smile:`.
Did you try it?

**Comment by renatobecker** [comment]
@knrt10,

There is a new property returned on the `config` method providing a list of RC custom emojis:

```"resources": {
"sound": "http://localhost:3000/sounds/chime.mp3",
"emojis": [
    {
        "_id": "3HN5spmYnmiijj4Jq",
        "name": "slight_smile",
        "aliases": [
            "slight_smile;slight_aa"
         ],
        "extension": "png",
        "_updatedAt": "2019-05-03T13:12:55.322Z"
    }
],
},```

So, we'll need to deal with this kind of emoji as well and get it from the server when necessary.

File: src/components/Messages/MessageText/index.js [hunk 1]
```diff
@@ -11,7 +12,7 @@ export const MessageText = memo(({
 }) => (
 	<div
 		// eslint-disable-next-line react/no-danger
-		dangerouslySetInnerHTML={{ __html: renderMarkdown(text) }}
+		dangerouslySetInnerHTML={{ __html: emojify(renderMarkdown(text), { output: 'unicode' }) }}
```
Comment (renatobecker) [inline src/components/Messages/MessageText/index.js]: Why not applying the `emojiify` stuff into `renderMarkdown` method?

Comment (knrt10) [reply src/components/Messages/MessageText/index.js]: hmm, that would be better

**Comment by renatobecker** [comment]
@knrt10, please follow the Tasso's instructions(last comment) and, once you ensure that everything is working fine, you need to add the message emoji rendering on Storybook as well.

**Comment by knrt10** [comment]
@tassoevan I am sorry I did not get it. Did you mean to check compatibility with Preact I should check in Storybook?

**Comment by tassoevan** [comment]
@knrt10 Let me clarify this: since Storybook supports only React, to use it we have to bridge our components via `preact-compat`. The final client bundle, however, doesn't have this compatibility layer, and things that are supposed to work over React only will work only on Storybook. You need to run the client, via `yarn start`, to check if the dependencies work over Preact alone.

**Comment by knrt10** [comment]
Thanks @tassoevan got it. I ran `yarn start` it worked fine. But to be clear should I run in production i.e `NODE_ENV=production` inside `.env` file or just run as it is? I tried both ways it's working fine

File: src/components/Messages/MessageText/customEmoji.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { store } from '../../../store';
+import { Livechat } from '../../../api';
+import render from 'preact-render-to-string';
+import { emojify } from 'react-emojione';
+
+function search(nameKey, myArray) {
```
Comment (renatobecker) [inline src/components/Messages/MessageText/customEmoji.js]: Replace to arrow function.
`const search = (key, array = []) => {...`

File: src/components/Messages/MessageText/customEmoji.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	for (let i = 0; i < myArray.length; i++) {
```
Comment (renatobecker) [inline src/components/Messages/MessageText/customEmoji.js]: Replace to `find` approach:
`return array.find((item) => item.name === key);`

Comment (renatobecker) [reply src/components/Messages/MessageText/customEmoji.js]: You need to check the `aliases` property as well.

File: src/components/Messages/MessageText/customEmoji.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	}
+}
+
+export const customEmoji = (text) => {
+
+	const { emojis } = store.state;
+	text = emojify(text, { output: 'unicode' });
+	const message = text.split(/:/g);
+	const content = [];
+	for (let i = 0; i < message.length; i++) {
```
Comment (renatobecker) [inline src/components/Messages/MessageText/customEmoji.js]: You can create an `arrow function` const and assign its return to `content` const.

File: src/components/Messages/MessageText/markdown.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
 import MarkdownIt from 'markdown-it';
-
+import { customEmoji } from './customEmoji';
```
Comment (renatobecker) [inline src/components/Messages/MessageText/markdown.js]: Let's replace the `customEmoji` name by `customEmojify`.

**Comment by knrt10** [comment]
Current status of picker.

Emoji picker

**Comment by tassoevan** [comment]
@sampaiodiego Any remarks about EmojiOne here?

**Comment by renatobecker** [comment]
Livechat already supports emoji rendering, so I'm closing this PR.

**Pull request closed without merging**